sdk/azcore/request.go

```diff
@@ -131,22 +131,20 @@ func (req *Request) MarshalAsByteArray(v []byte, format Base64Encoding) error {
 }
 
 // MarshalAsJSON calls json.Marshal() to get the JSON encoding of v then calls SetBody.
-// If json.Marshal fails a MarshalError is returned.  Any error from SetBody is returned.
 func (req *Request) MarshalAsJSON(v interface{}) error {
 	v = cloneWithoutReadOnlyFields(v)
 	b, err := json.Marshal(v)
 	if err != nil {
-		return fmt.Errorf("error marshalling type %s: %w", reflect.TypeOf(v).Name(), err)
+		return fmt.Errorf("error marshalling type %T: %s", v, err)
```

Comment:
Why is %s now used for err? Should it then be %s for err.Error()?

Reply:
Per https://golang.org/pkg/fmt/

4. If an operand implements the error interface, the Error method will be invoked to convert the object to a string, which will then be formatted as required by the verb (if any).